tests/src/Kernel/Entity/OgSelectionTest.php

```diff
@@ -138,28 +165,45 @@ public function testSelectionHandler() {
    * versa.
    */
   public function testSelectionHandlerResults() {
-    $user1_groups = $this->createGroups(2, $this->user1);
-    $user2_groups = $this->createGroups(2, $this->user2);
+    $user1_groups = $this->createGroups(5, $this->groupAdmin);
+    $user2_groups = $this->createGroups(5, $this->groupManager);
 
-    // Check that users get the groups they manage.
-    $this->setCurrentAccount($this->user1);
+    $all_groups_ids = array_merge($user1_groups, $user2_groups);
+
+    // Admin user can create content on all groups.
+    $this->setCurrentAccount($this->groupAdmin);
     $groups = $this->selectionHandler->getReferenceableEntities();
-    $this->assertEquals($user1_groups, array_keys($groups[$this->groupBundle]));
+    $this->assertEquals($all_groups_ids, array_keys($groups[$this->groupBundle]));
 
-    $this->setCurrentAccount($this->user2);
+    // Group manager can create content in their groups.
+    $this->setCurrentAccount($this->groupManager);
     $groups = $this->selectionHandler->getReferenceableEntities();
     $this->assertEquals($user2_groups, array_keys($groups[$this->groupBundle]));
 
-    // Check the other groups.
-    $this->selectionHandler = Og::getSelectionHandler($this->fieldDefinition, ['handler_settings' => ['field_mode' => 'admin']]);
+    // Non-group member.
+    $this->setCurrentAccount($this->groupMember);
+    $groups = $this->selectionHandler->getReferenceableEntities();
+    $this->assertTrue(empty($groups[$this->groupBundle]));
 
-    $this->setCurrentAccount($this->user1);
+    // Group member access to create content.
+    $group_id = $user1_groups[0];
+    $group = Node::load($group_id);
+    $membership = Og::createMembership($group, $this->groupMember);
+    $membership->save();
+
+    // Group member cannot create content in their groups when they don't have
+    // access to.
     $groups = $this->selectionHandler->getReferenceableEntities();
-    $this->assertEquals($user2_groups, array_keys($groups[$this->groupBundle]));
+    $this->assertTrue(empty($groups[$this->groupBundle]));
+
+    // Grant OG permission.
+    $og_role = OgRole::getRole('node', $this->groupBundle, OgRoleInterface::AUTHENTICATED);
+    $og_role
+      ->grantPermission('create node ' . $this->groupContentBundle)
```

Comment:
We should fix this in #510:
```suggestion
      ->grantPermission("create {$this->groupContentBundle} content")
```